### PR TITLE
Adding text in section 6.4.4.3 (ident element) 

### PR DESCRIPTION
**Pull request by bgro**
Proposal for treating issue  https://github.com/scapcommunity/XCCDF/issues/12

### CODE REVIEW

**Comment by djhaynes** [comment]
This looks reasonable.

**Comment by DavidRies** [comment]
Generally, this approach looks good to me. I have two suggestions:

1. There should have a way to identify when a specific `<xccdf:ident>` is being used as a lifetime resource identifier ("LRI")
2. I think the @historic attribute and values could be more clearly named

Here are a couple alternate ideas that would accomplish the above:
```
<!-- Option A: use "urn:lri:.." to indicate this is an LRI and @rel to indicate relation -->
<ident system="urn:lri:jovalcm.com">LRI-112</ident>
<ident system="urn:lri:jovalcm.com" rel="superseded">LRI-131</ident>
<ident system="urn:lri:jovalcm.com" rel="related">LRI-113</ident>

<!-- Option B: use @lri to indicate this is an LRI and express relation -->
<ident system="http://jovalcm.com/xccdf-lri" lri="true">LRI-112</ident>
<ident system="http://jovalcm.com/xccdf-lri" lri="superseded">LRI-131</ident>
<ident system="http://jovalcm.com/xccdf-lri" lri="related">LRI-131</ident>

**Comment by bgro** [comment]
Of @DavidRies suggestions, I would prefer the first one, namely defining a definite syntax for SCAP rule LRI systems, `urn:lri:<issuer_identifier>` or maybe better `urn:xccdf:lri:<issuer_identifier>`: that would make it easier for tooling. 

Also I am happy to use a `rel` attribute ... maybe worth taking a few more brain cycles to be sure that the list of allowed values for that attribute is complete.

I will wait a bit whether there are more comments and would then update the proposal in time for the meeting in three weeks' time.

**Comment by redhatrises** [comment]
Reading this, I would like to see the verbiage and wording updating to match the rest of the spec. A table would be nice to add.

**Comment by bgro** [comment]
I updated the CR taking into account David's comments. As part of this, I tried to come up with what might be useful attributes for describing a relationship. Please have a look and tell me whether this is overengineering and we should stay with `superseded` and `related`:


   | Value of relation attribute | Intended semantics  |
   | --- | --- |
   | superseded | The specified LRI used to refer to this rule in a previous version of the baseline |
   | formerly_part_of | This rule defines an aspect which used to be treated (amongst other aspects) within the referenced rule |
   | contains | Among other aspects, this rule contains an aspect that used to be treated in the referenced rule |
   | refactored_from | This rule originiated by refactoring the referenced rule, but the nature of the relationship cannot be described properly with `formerly_part_of` or `contains` |
   | related | This rule is related to the referenced rule in a previous version of the baseline, but none of the above values precisely capture this relationship |

**Comment by DavidRies** [comment]
Hello Bernd,

This looks good. Do these examples accurately capture what you had in mind?

**@rel=superseded**: the LRI for Rule-1 is changed, but the rule has not been refactored to/from other rules
```
Rule-1
    - current Rule-1 LRI
    - former Rule-1 LRI with @rel=supersede
```

**@rel=formerly_part_of**: Rule-1 refactored into Rule-2 and Rule-3:
```
Rule-2
    - current LRI for Rule-2
    - Rule-1 LRI with @rel=formerly_part_of
Rule-3
    - current LRI for Rule-3
    - Rule-1 LRI with @rel=formerly_part_of
```

**@rel=contains**: Rule-1 and Rule-2 are refactored into Rule-3:
```
Rule-3
    - current LRI for Rule-3
    - Rule-1 LRI with @rel=contains
    - Rule-2 LRI with @rel=contains
```

**@rel=refactored_from**: parts of Rule-1 and Rule-2 are refactored into parts of Rule-3 and Rule-4:
```
Rule-3
    - current LRI for Rule-3
    - Rule-1 LRI with @rel=refactored_from
    - Rule-2 LRI with @rel=refactored_from
Rule-4
    - current LRI for Rule-4
    - Rule-1 LRI with @rel=refactored_from
    - Rule-2 LRI with @rel=refactored_from
```

**Comment by bgro** [comment]
Yes, exactly, thanks for spelling out these examples! 

Like I said, I am not sure whether this is overengineering or not and might maybe be contraproductive because it might put issuers off providing relationship information. It that were the case, then I'd be happy with 'superseded' and 'related' and would take the time to figure out the rest by myself when such cases occur...

**Comment by bgro** [comment]
David just made the suggestion of sticking with `related` and `superseded` and allow an additional comment. 

Furthermore the question is whether these relationships info should stay in there for ever, be removed for the next-plus-one version of the baseline, etc. We need some guidance for that in the text. 

Finally, discussion on what substantive changes are (CIS is working with such a concept for managing the tailoring in the workbench).

Also: should the lrn-system be used to relate back to rules in the benchmark of a different publisher, or should it only be used to relate back to LRIs of the same issuer?

Eg, when building upon a benchmark from CIS, should it be

~~~
<ident system="urn:xccdf:lri:cisecurity.org" rel="related">19-12454</ident>
<ident system="urn:xccdf:lri:siemens.com">BL968-7342</ident>
~~~

or should it be something like

~~~
<ident system="urn:xccdf:lri-reference>cisecurity.org:19-12454</ident>
<ident system="urn:xccdf:lri:siemens.com">BL968-7342</ident>
~~~

The first option is probably easier, even though it broadens the use of the lri system from just providing references to own stuff to referencing XCCDF rules in general... that again might have more use-cases which might ask of other `relation` attribute values (`ident`, `derived_from` ...)

**Comment by DavidRies** [comment]
A few additional thoughts, feedback.

**Using LRIs for Relations Between Content from Different Publishers**
It seems to me that LRIs should be scoped to a Benchmark and not a publisher, i.e. we should make it explicit that an LRI's "lifetime" is the lifetime of a Benchmark with a specific Benchmark@id. So, by "all versions of a Benchmark" we mean all Benchmarks with the same Benchmark@id. Benchmarks could include LRIs from different orgs if, for example, multiple orgs were writing it collaboratively. But, LRI's are not intended for tracking relationships between rules in different Benchmarks whether they are from other publishers or not (i.e. between a Windows 2016 STIG and a CIS Windows Benchmark or between a CIS Windows 2016 Benchmark and a CIS Windows 2019 Benchmark).

**`urn:xccdf:lri:cisecurity.org` vs `cisecurity.org:19-12454`**
I prefer the first of the two options above (`<ident system="urn:xccdf:lri:cisecurity.org" rel="related">19-12454</ident>`) in which the system URI serves two functions (identifying this a URI and establishing a namespace so the ns owner can manage IDs) and the ID is a simple string ID. This seems simpler and more consistent with the use of similar attributes and values in XCCDF.

**@benchmark_id**
Should we add an optional `@benchmark_id` that defaults to the value of the document's Benchmark@id? I know I just advocated for limiting the scope of LRIs to a specific Benchmark@id, but there are some limited exceptions to this rule that stay in the spirit of an LRI such as: an publisher changes the Benchmark@id for some reason; or, an org makes a custom "downstream" version of a Benchmark including changes to Rules and the Benchmark@id.

**Should LRI@rel remain in document forever or just for one version?**
In order to use an LRI with `@rel` you will need to have the previous version(s) of the Benchmark containing the related LRI. So, if you need to have the previous versions anyway, you will have the LRIs in the previous version as well which, IMO, means we don't need to keep LRI relations for more than 1 version. We could make it explicit that LRIs with @rel should only indicate a relation to the LRIs in the immediately preceding Benchmark version. That would reduce document bloat and make LRIs more immediately useful, IMO.

**Comment by djhaynes** [comment]
Sorry, I missed this in the chat for our 7/9 meeting, but, Josh said:

As an alternative to LRIs, what about adding a changelog child element to xccdf:Benchmark? This could be implemented via a script to convert a GitHub change log into the xccdf:changelog format.

**Comment by wmunyan** [comment]
I volunteered to document here, what CIS has been documenting internally on what constitutes a "substantive" change, which would cause the generation of a new LRI for a recommendation.  For us and our authoring tooling (CIS WorkBench, hereafter abbreviated as WB), our challenge is that we essentially have 2 stakeholders creating content:  internally our benchmark development team is creating benchmarks and revising them over time, and externally members have the ability to "fork" benchmarks, tailoring them to their specific needs.

For our internal benchmark development, the easiest workflow is the creation of a new benchmark.  In this case, new recommendations are created and edited in a draft status.  Once the benchmark has gone through our consensus process and published, the recommendations are considered final and new LRIs generated.  When the benchmark developer is revising that content for "v-next", recommendations may be updated prior to publishing that next version.  When editing a recommendation, the benchmark developer will need to make the decision on whether or not the change to the recommendation is substantive.  Because we are very early in this process of implementation in WB, we are still ironing out the "rules" for what constitutes a substantive change, other than to say that "if the intent of the recommendation is changing, its a substantive change".  Edits such as fixing typos, changes to the description or rationale, etc are _not_ substantive.  Even changes to how a recommendation has system characteristics collected/evaluated are not substantive; a new collection method may have been created i.e. `userrights_test` vs `accesstoken_test`.  The short version is that currently the idea is that it is a manual decision that each benchmark developer will have to make, determining if a modification is substantive, and to therefore generate a new LRI.

In the "fork"/tailoring case, the only difference is in the initial forking process and edit.  When a member decides to tailor a CIS benchmark, they begin with an existing published version, and "fork" it (i.e. make a copy).  In this case, the initial forked version of the benchmark is considered "locked", and _any_ change made to that initial "fork" is automatically considered substantive, thus generating a new LRI.  Once the member finalizes this initial "fork" of the benchmark and moves on to revisions of that content, it follows the same rules for "v-next" as described above for CIS benchmark development.

**Comment by bgro** [comment]
@wmunyan :  You write: "if the intent of the recommendation is changing, its a substantive change"
I seem to remember that changing the prescribed password length would not be considered changing the intent. So I am struggling with coming up with examples that CIS might consider changing the intent.

Could you supply some examples of what would be changes that change the intent of a rule? This really would be helpful for understanding your approach!

**Comment by wmunyan** [comment]
@bgro I think here at CIS we're still debating on that as well.  I went back to a colleague who worked with me internally on a design and we seem to have differing opinions on what we discussed.

An example could be the following.  Imagine an initial recommendation which specifies to "Ensure the NTP service is running".  That recommendation is going to be modified to "Ensure NTP or chrony service is running".  In this case, the recommendation changing from ensuring a single service is running to one of two services is running modifies the intent, and would therefore be considered substantive.

I am open to discussion on any of the other types of modifications to be considered substantive or not.  For example, changing the requirement of password length from 14 to 20 may or may not be substantive.  Modifying a password recommendation from checking `pam_cracklib` to checking `pam_pwquality` may or may not be substantive.  We're still early in our design and havent yet begun implementation so I'd definitely like to include community consensus in the design.

**Comment by bgro** [comment]
I see the following open issues:

1) **CIS is working on a concept of substantive vs. non-substantive change.**

    The concept is not yet completely clear and is not reflected in the current text in #28, which 
    provides the following exceptions for changing a lifetie identifier:

    ~~~
    Exceptions may occur in the following cases:

    a. Keeping the reference identifier unchanged would lead to
       confusion (e.g., because the reference identifier carries
       semantics and a rule change would lead to a contradiction
       between what a user would understand when looking at the
       reference identifier and the actual contents of the rule.)

    b. There is no clear one-to-one relationship between the old
       version of a rule and a corresponding rule in the new version of
       the baseline (e.g., because a rule has been re-factored into many
       rules.)
    ~~~

    This text would also keep the life-time identifier for `substantive` changes:
    as long as the author considers the new version of the rule as having replaced
    the previous rule, the life-time identifier wouuld have to stay the same.

2) **Currently, the proposal defines a somewhat fine-grained relationship attribute:**

   Value of relation attribute | Intended semantics  
   --- | --- 
   superseded | The specified LRI used to refer to this rule in a previous version of the baseline 
   formerly_part_of | This rule defines an aspect which used to be treated (amongst other aspects) within the referenced rule 
   contains | Among other aspects, this rule contains an aspect that used to be treated in the referenced rule 
   refactored_from | This rule originiated by refactoring the referenced rule, but the nature of the relationship cannot be described properly with `formerly_part_of` or `contains` 
   related | This rule is related to the referenced rule in a previous version of the baseline, but none of the above values precisely capture this relationship 

   It seems that there is a tendency to restrict this to `superseded`, `related` (and maybe for formal reasons, we need `ident` as default which can be provided explicitly
   or is implicit if there is no `rel` attribute)

3) **Do we need guidance on how long LRI-backlinks with `superseded` or `related` should be kept around?**

   Possible alternatives:

   a) say nothing
   b) say they should be carried over also in subsequent versions of a baseline  
   c) say they should be present only in one version and removed in the next (minor) version

   I tend towards *c*

4) **Specify more clearly scope of LRI**
   
   In a comment, David writes:

   ~~~
   Using LRIs for Relations Between Content from Different Publishers

   It seems to me that LRIs should be scoped to a Benchmark and not a publisher, i.e. we should make it explicit that an 
   LRI's "lifetime" is the lifetime of a Benchmark with a specific Benchmark@id. So, by "all versions of a Benchmark" we 
   mean all Benchmarks with the same Benchmark@id. Benchmarks could include LRIs from different orgs if, for example, 
   multiple orgs were writing it collaboratively. But, LRI's are not intended for tracking relationships between rules 
   in different Benchmarks whether they are from other publishers or not (i.e. between a Windows 2016 STIG and a CIS 
   Windows Benchmark or between a CIS Windows 2016 Benchmark and a CIS Windows 2019 Benchmark).
   ~~~

   I do not agree about different benchmarks of the same publisher : it is common practice to
   derive a benchmark for a newer release from an older release, e.g., derive Windows Server 2019 from Windows Server 2016.
   Having LRIs carry over, either without any changes (in that case we probably would consider the benchmark ID implicitly
   as part of the LRI) or with changes and backlinks via the `rel='superseded` attribute would be very useful for consumers.

   As for different publishers: I also would tend leave LRIs. Here, it is important to note that an LRI has
   two components: the system (which includes a reference to the publisher) and the LRI "itself".
 
   Consider the use-case of CIS and the forks on the workbench:

   As soon as a benchmark is forked, the LRI-system would be changed  to include an identifier
   of the organization owing the fork rather than CIS, but leave the LRI unchanged. At the same time, a back reference 
   with `rel='related'` would be added to the LRI in the CIS-LRI system.

   Thus, when the fork is published, the relationship to the original rules is clear *and* the customary LRI can be
   used (if I have come to know that "BL1234" is about password length, then I will have the same in my forked baseline (if I want)

   So, in short: when modifying a benchmark, I would allow keeping the LRI (as long as the system is changed) if the 
   organization doing the modifying wants that. I would strongly suggest (SHOULD), that a backward reference with `rel='related` 
   is added in the first published version of a modified baseline.
 

5) **Is LRI relative to the benchmark ID?**

   David mused about this in a comment above.

   My interpretation is that the URI of a rule is created by combining somehow the ID of the baseline with the LRI of the rule.
   Only then can CIS have the same LRI for, say, the password-length rule, both in Windows Server 2016 and Windows Server 2019.

   The URI for linking to the rule must be different for the two baselines, because the latest version of the rule in Server 2016
   could be different to a rule of the same LRI in Server 2019.

   So, the LRI itself  would have two different uses, depending on with what it is combined:

   - combine LRI with LRI system: denotes either exactly one rule which never has changed or a "cluster" of rules which are 
      interrelated by a version tree (tree, because the rule may evolve differently in different benchmarks)

   - combine LRI with id of benchmark: a link to the latest version of that rule as given in the latest version
     of that benchmark.

**Comment by bgro** [comment]
As agreed in the workgroup meeting of 2020-07-30, here is a list of concrete questions, for which we require input

# *Q1* **How fine-grained should the relationship attribute be**

## Possible Answers

Please answer: 
 
A) Only `superseded` and `related`
B) All values given in the table below
C) A subset of values / different values (please specify)

## Background Information

Currently, the proposal defines a somewhat fine-grained relationship attribute:

Value of relation attribute | Intended semantics  
--- | --- 
superseded | The specified LRI used to refer to this rule in a previous version of the baseline 
formerly_part_of | This rule defines an aspect which used to be treated (amongst other aspects) within the referenced rule 
contains | Among other aspects, this rule contains an aspect that used to be treated in the referenced rule 
refactored_from | This rule originiated by refactoring the referenced rule, but the nature of the relationship cannot be described properly with `formerly_part_of` or `contains` 
related | This rule is related to the referenced rule in a previous version of the baseline, but none of the above values precisely capture this relationship 

It seems that there is a tendency to restrict this to `superseded`, `related` 

## My (Bernd) two cents

I favor **A**, because readers who wish to understand exactly what has changed will have to compare the rules anyhow: the main point of the proposed mechanism is to point them to the right rule with which the comparison is to be carried out.

# *Q2* **What kind of guidance should be given about how long back-references to changed LRIs should be kept**

## Possible Answers

A) The back reference should only be present in the revision of the rule/baseline in which the change of the LRI occured
B) The back reference should be kept indefinitely
C) The back reference should be kept until the publisher deems that  does not make sense anymore (e.g., because
    the change of the LRI occured a really long time ago).
D) Something else (please specify)

## My (Bernd) two cents

I am against *A*, because it would mean that if I "miss" the moment at which an LRI changes, because I did not
look at the revision of the baseline/rule where the change occured, then I would need to hunt through all intermediate
revisions to find out whether there was a change and what the original LRI was. So *A* is out.

There remain *B* and *C*. *B* is more precise, but I can imagine situations where a back reference really does
not make sense any more and thus would not want to force the publisher to carry senseless reference around
indefinitely.

So: I tend towards *C*

# *Q3* **Should there be a special mechanism to qualify an LRI (e.g., as being relativve to a  specific baseline?)**

## Possible Answers

A) If publishers want to restrict the scope of the LRI, they should design the LRI form such that they can do so
    (e.g., have the LRI in form "BL123-1454" where the `123` is a unique number used by the publisher for its baselines)
B) There should be an additional (optional) attribute carrying scope information, maybe with a special
   mechanism denoting that the current baseline is the scope
C) There should be some kind of other special mechanism

## Background information

An obvious usage of an LRI is to generate a URI/L from it, which allows a user to access the rule via the web, an API...

If you are an organization such as Compliance-as-Code, where one and the same rule is shared in various
baselines but managed at a single location, an LRI without scope information would be enough to always
find the latest version of that rule.

If, on the other hand, you are building new baselines by copying the content of an older baseline (e.g. going
from Windows Server 2016 to 2019), then you either have to change the LRI to be able to use it
for deriving a URI, or you need a mechanism to qualify the LRI with scope (e.g., the Server 2016 vs Server 2019
baseline). The URI would then be derived from the LRI-system, the LRI-scope and the LRI itself.


## My (Bernd) two cents

I favor *B*, because it is the most flexible and suits both organization such as Compliance-as-Code (who would not use it) and organizations that copy content when creating new baselines. 

# *Q4* **What are changes to a rule, which demand the change of an LRI**

## Possible Answers

Please provide instances (CIS calls them `substantive` changes) which in your opinion would require the change of an LRI.
Think of

- changes to the content of the rule
- changes regarding the baseline (e.g., a baseline for a new OS release is issued based on the baseline of an existing OS release)
- changes regarding rule ownership (e.g., when forking a baseline in the CIS workbench)

## Background Information

 the current text in #28, which 
 provides the following exceptions for changing a lifetie identifier:

 ~~~
   Exceptions may occur in the following cases:

    a. Keeping the reference identifier unchanged would lead to
       confusion (e.g., because the reference identifier carries
       semantics and a rule change would lead to a contradiction
       between what a user would understand when looking at the
       reference identifier and the actual contents of the rule.)

    b. There is no clear one-to-one relationship between the old
       version of a rule and a corresponding rule in the new version of
       the baseline (e.g., because a rule has been re-factored into many
       rules.)
 ~~~

This text would also keep the life-time identifier for `substantive` changes:
as long as the author considers the new version of the rule as having replaced
the previous rule, the life-time identifier would have to stay the same.

But currently , the specification lacks guideance on what to do when a baseline
for a new release of an OS/application is published or when an ownership change occurs.

## My (Bernd) two cents

*Regarding changes to contents of a rule*:

As given above: as long as the author considers the new version of the rule as having replaced
the previous rule, the life-time identifier would have to stay the same. Even if the new rule
says the complete opposite of the previous version of a rule (e.g., it goes from "Use 15 letter passwords"
to "Use an empty password") the LRI would have to stay the same, because it is the same setting
we are talking about, namely password length. 

Now, if the new baseline has a rule "Do not use passwords at all",
then I would say that this is a completely new rule with a new LRI, which does no require a back-reference
to the old rule about password length (which will have been removed in the new version of the baseline.

*Regarding reuse of  rules in a new baseline*

There may be  a case for changing the LRI in case what was once the same rule evolves in the context
of one baseline and does not evolve (or evolves in a different direction) in the context of another baseline.
An organization may want to assign a unique LRI to different variants of a rule which derived from a common
ancestor but have no clear "old-vs-new" relationship.

Whether an LRI is changed "on demand" when the situation above of diverging development occurs,
or LRIs are changed whenever a rule is used in a different context (always providing a back reference)
depends on how the publisher manages rules: are rules shared between several baselines (as 
is the case for Compliance as Code) or are they copied and given a life of their own when
they are reused in a new baseline?

Eg., at Siemens, our LRIs are structured as "BL123-4567", where 123 denotes the baseline and 4567 a rule in that baseline.
If we reuse the same rule in a different baseline, we would keep the "4567" and change the "123" part so as to
reflect the new baseline So here, the LRI changes, but it changes in a way that makes the relationship between the different rules
explict to the reader.

So I think we should add guidance that an LRI MAY change if a rule is re-used, to reflect that this is a new
instance of the rule, possibly with its own path of evolvement separate from the rule from which it was copied. 
Another option would be to do without LRI changes would be answer *B* or *C* of *Q3*: change the LRI-scope
when re-using a rule.

*Regarding change of ownership*

The new owner is likely to have his own LRI system and thus would change the LRI (ideally providing
a back reference to the LRI of the former owner (see *Q5* below)). But for processes like forking
on the CIS workbench, the LRI could, at least for the start, stay the same: the fork would
change the LRI system of the LRI and provide a back reference to the old LRI with CIS' LRI system.


# *Q5* **Should backrefs only occur between a rules in different revisions of one and the same baseline**

## Possible answers

A) Backrefs should only occur between a rules in different revisions of one and the same baseline.
B) Backrefs can occur either between a rules in different revisions of one and the same baseline or,
    where required, between different baselines (e.g. Windows Server 2016 and Windows Server 2019)
    of the same publisher
C) Backrefs can occur either between a rules in different revisions of one and the same baseline or, 
    between different baselines (e.g. Windows Server 2016 and Windows Server 2019) of the same publisher and,
    where required, between baselines of different publishers/owners in the special case that the baseline
    of one owner was derived from a baseline of a different owner/publisher

## My (Bernd) two cents

I prefer *C*, because there are valid use-cases for all listed options:

It is common practice to derive a benchmark for a newer release from an older release, e.g., derive Windows Server 2019 from Windows Server 2016. Having LRIs carry over, either without any changes  or with changes and backlinks via the `rel='superseded` attribute would be very useful for consumers: most likely, the consumer already has implemented
the benchmark for Windows Server 2016 and now wants to know how the new benchmark for Windows Server 2019 relates to the old one for Server 2016.  

As for different publishers/owners:
 
Consider the use-case of CIS and the forks on the workbench:

As soon as a benchmark is forked, the LRI-system would be changed  to include an identifier
of the organization owing the fork rather than CIS, but leave the LRI unchanged. At the same time, a back reference 
with `rel='related'` would be added to the LRI in the CIS-LRI system.

Thus, when the fork is published, the relationship to the original rules is clear *and* the customary LRI can be
used (if I have come to know that "BL1234" is about password length, then I will have the same in my forked baseline (if I want)

So, in short: when modifying a benchmark, I would allow keeping the LRI (as long as the system is changed) if the organization doing the modifying wants that. I would strongly suggest (SHOULD), that a backward reference with `rel='related` 
is added in the first published version of a modified baseline.

**Comment by DavidRies** [comment]
Thanks for putting all this together! Here's my input:

Q1: A) Only superseded and related
_For the reasons mentioned by Bernd and because this seems the simplest to define/document and least ambiguous. Also, there should be an optional comment to allow a note about the change._
Q2: A) The back reference should only be present in the revision in which the change occurs
_This would reduce bloat and make it clear that each back reference is immediately relevant (indicates a change in the current version. If someone is doing change analysis/tracking across versions, they will need access to all previous versions anyway and should be able to find all the back references therein. To support this, we could add a benchmark version history element with a list of all previous versions of a specific benchmark._
Q3: B) There should be an additional (optional) attribute carrying scope information, maybe with a special
mechanism denoting that the current baseline is the scope
_This would give content authors the option of specifying relationships between rules in various benchmarks. If omitted, the default would be that the references applies to the current benchmark (i.e. current benchmark publisher/id but not necessarily the version)._
(Q4 skipped for now)
Q5: C)

**Comment by bgro** [comment]
Notes from telco on 2020-08-20 (updated on 2020-09-28)

- *Q1*: Consensus for Option 1 (with the expectation for publishers to really use the backwards references in case they change the LRI for some reason. I have updated the merge requets accordingly.

- *Q2*: No clear consensus, the two options seem to be A (which, e.g., David favours) and C (which would be my favourite)

- *Q3* and *Q5*: still unclear. Some argue for scoping always to a single benchmark, but the way Compliance as Code operates would suggest "publisher-wide" scope.  There also was a plea for keeping things simple -- this would mean that the scope must always be the same.  It seems to me that if that is what we want to achieve, then we need to pick "publisher-wide" scope as default and require the publishers who want to scope rules with regards to a specific baseline to include a reference to the baseline within the LTI (e.g, have the LTI be of form `<document-prefix>-<rule-suffix>`.

- *Q4*: it was my understanding that the participants can live with the current wording:

  ~~~
    Exceptions may occur in the following cases:

   a. Keeping the reference identifier unchanged would lead to
      confusion (e.g., because the reference identifier carries
      semantics and a rule change would lead to a contradiction
      between what a user would understand when looking at the
      reference identifier and the actual contents of the rule.)

   b. There is no clear one-to-one relationship between the old
      version of a rule and a corresponding rule in the new version of
      the baseline (e.g., because a rule has been re-factored into many
      rules.)
  ~~~

So, for further processing by NIST, this would mean:

- the proposed changes reflect the community consensus on Q1 and Q4
- additional guidance has to be added regarded the final design decision taken on Q2, Q3, and Q5.
